src/docfx/abstractions/ITemplateEngine.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal interface ITemplateEngine
+    {
+        bool IsPage(string schemaName);
+
+        bool TryGetLocalizedToken(string token, string localizedToken);
+
+        JsonSchema GetJsonSchema(SourceInfo<string> schemaName);
```

Comment:
Split it into IJsonSchemaProvider, shared it with IJsonSchemaValidator and IJsonSchemaTransformer?